gdal/frmts/vrt/pixelfunctions.cpp

```diff
@@ -915,6 +924,79 @@ static CPLErr PowPixelFunc( void **papoSources, int nSources, void *pData,
 
 }
 
+// Given nx intervals spaced by dx and beginning at x0, return the the index of
+// the lower bound of the interval that should be used to interpolate/extrapolate
+// a value for x.
+static std::size_t intervalLeft(double x0, double dx, std::size_t nx, double x)
+{
+    if (x < x0) {
+        return 0;
+    }
+
+    std::size_t n = static_cast<std::size_t>((x - x0) / dx);
```

Comment:
there's a slight chance that the cast might be invalid in case of dx being very close to 0

Reply:
I can check for that and handle it. Are you opposed to pixel functions throwing exceptions that are caught and converted to CPL errors at the single call site? I think that would keep the error handling cleaner.

Reply:
>  that are caught and converted to CPL errors at the single call site?

sounds good